src/util/include/util/serializing/ObjectInputStream.h

```diff
@@ -52,4 +54,43 @@ class ObjectInputStream {
     size_t len = 0;
 };
 
-extern template void ObjectInputStream::readData(std::vector<double>& data);
+// This function requires that T is read from its binary representation to work (e.g. integer type)
+template <typename T>
+T readTypeFromSStream(std::istringstream& istream) {
+    if (istream.str().size() < sizeof(T)) {
+        std::ostringstream oss;
+        oss << "End reached: trying to read " << sizeof(T) << " bytes while only " << istream.str().size()
```

Comment:
Same here, copy of string

Reply:
I have removed all use of `istream.str()`